pkg/utils/helper.go

```diff
@@ -11,6 +12,64 @@ import (
 	"strings"
 )
 
+// PublicAzureStorageEndpointSuffix defines default Storage Endpoint Suffix
+const PublicAzureStorageEndpointSuffix = "core.windows.net"
+
+// Azure defines Azure configuration
+type Azure struct {
+	Cloud string `json:"cloud"`
+}
+
+// AzureStackCloud defines Azure Stack Cloud configuration
+type AzureStackCloud struct {
+	StorageEndpointSuffix string `json:"storageEndpointSuffix"`
+}
+
+// IsAzureStackCloud returns true if the application is running on Azure Stack Cloud
+func IsAzureStackCloud() bool {
+	azureFile, err := RunCommandOnHost("cat", "/etc/kubernetes/azure.json")
+	if err != nil {
+		return false
+	}
+	var azure Azure
+	err = json.Unmarshal([]byte(azureFile), &azure)
+	if err != nil {
+		return false
+	}
+	cloud := azure.Cloud
+	return strings.ToUpper(cloud) == "AZURESTACKCLOUD"
+}
+
+// CopyCert saves the cert content to a new file inside the container
+func CopyCert() error {

```

Comment:
Why do we need to do this?

Reply:
If we don't, we get this error: 

`Collector: systemperf, export data failed: Failed to create blob: -> github.com/Azure/azure-pipeline-go/pipeline.NewError, C:/Users/t-smurali/Desktop/aks-periscope/vendor/github.com/Azure/azure-pipeline-go/pipeline/error.go:154
HTTP request failed
Put "https://shweta.blob.redmond.ext-n31r1205.masd.stbtest.microsoft.com/test-cluster1-redmond-cloudapp-ext-n31r1205-masd-stbtest-micros?restype=container&se=2021-08-07T07%3A22%3A25Z&sig=JlwqovhJdorwmnFudse": x509: certificate signed by unknown authority`

It is because we need the self signed cert in order to communicate with the storage account.

Reply:
Great! May I suggest to explain briefly this in the code so that we know why azure stack requires this?